### PR TITLE
Add artifact created event based on the outbox pattern

### PR DESCRIPTION
**Pull request by carlesarnal**
This PR introduces support for having Registry as a source for events. In the case of the SQL storage, an outbox table has been added and a row is created for each of the following events:

- Group Created
- Group Metadata Updated
- Group Deleted
- Artifact Created
- Artifact Metadata Updated
- Artifact Deleted
- Artifact Version Created
- Artifact Version Metadata Updated
- Artifact Version Deleted
- Global Rule Configured
- Group Rule Configured
- Artifact Rule Configured

Note that the event sending is done at the storage impl level on purpose, so we have access to the underlying transaction so we can benefit from the usual advantages of the outbox pattern in the SQL storage. We don't have now a way to intercept the transaction, so this was the only possible way.

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlFactory.java [hunk 1]
```diff
@@ -54,6 +54,11 @@ public class KafkaSqlFactory {
     @Info(category = "storage", description = "Kafka sql snapshots store location")
     String snapshotStoreLocation;
 
+    @Inject
+    @ConfigProperty(name = "apicurio.events.kafka.topic", defaultValue = "registry-events")
+    @Info(category = "storage", description = "Kafka sql storage event topic")
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlFactory.java:59]: Isn't there some kind of `sinceVersion` property we can configure for this?

Comment (EricWittmann) [reply app/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlFactory.java:59]: Same with any other config property introduced in this PR....

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java [hunk 2]
```diff
@@ -3524,10 +3583,44 @@ public String createSnapshot(String location) throws RegistryStorageException {
...
+        }
+        return event.getId();
+    }
+
+    @Override
+    public boolean supportsDatabaseEvents() {
+        return isPostgresql() || isMssql();
+    }
+
+    private void deleteEvent(String eventId) {
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java:3605]: I tried to have a pattern in the SQL storage where only public methods would call `handles.withHandle()` - the resulting `handle` instance would get passed as an argument to private methods needing it.  Could you change this to follow that pattern?

**Review by EricWittmann — CHANGES_REQUESTED** [review]
Looks good.  Nice testing.  I had a couple minor comments.

**Review by EricWittmann — APPROVED** [review]
LGTM as-is.  However, question for you:

Do we ever want to offer the option of SQL storage variant + Kafka eventing **without** using Debezium?

**Comment by carlesarnal** [comment]
> LGTM as-is. However, question for you:
> 
> Do we ever want to offer the option of SQL storage variant + Kafka eventing **without** using Debezium?

Good question. If the Kafka cluster is there anyway, what would be the reason to not use Debezium? In the end, what we're doing is very basic CDC, so I don't see any benefits other than simplifying (a bit) the deployment model. For now, I'll merge as-is, we can revisit later if we think it makes sense.

**Pull request merged**